pilot/pkg/model/push_context.go

```diff
@@ -2061,6 +2061,10 @@ func NewGatewayContext(ps *PushContext) GatewayContext {
 	return GatewayContext{ps}
 }
 
+func (gc GatewayContext) GetService(hostname string) *Service {
+	return gc.ps.ServiceForHostname(nil, host.Name(hostname))
```

Comment:
can we pass hostname + namespace to this method? As is it may give inaccurate results in some cases.